### PR TITLE
Logout for NoPW users, frontend changes

### PR DESCRIPTION
**Pull request by zapu**

### CODE REVIEW

File: go/service/user.go [hunk 1]
```diff
@@ -628,8 +628,9 @@ func (h *UserHandler) CanLogout(ctx context.Context, sessionID int) (res keybase
 
 	if hasRandomPW {
 		return keybase1.CanLogoutRes{
-			CanLogout: false,
-			Reason:    "You signed up without a password and need to set a password first.",
+			CanLogout:     false,
+			SetPassphrase: true,
+			Reason:        "You signed up without a password and need to set a password first.",
```
Comment (buoyad) [inline go/service/user.go]: 👌

File: shared/actions/config/index.js [hunk 2]
```diff
@@ -245,6 +247,31 @@ const switchRouteDef = (state, action) => {
 
 const resetGlobalStore = () => ({payload: null, type: 'common:resetStore'})
 
+// Figure out whether we can log out using CanLogout, if so,
+// startLogoutHandshake, else do what's needed - right now only
+// redirect to set passphrase screen.
+function* startLogoutHandshakeIfAllowed(state) {
```
Comment (buoyad) [inline shared/actions/config/index.js]: Use `chainAction` for this:
```
const startLogoutHandshakeIfAllowed = state => 
    RPCTypes.userCanLogoutRpcPromise().then(canLogoutRes => {
        // return action or array of actions here
    })

File: shared/settings/passphrase/container.js [hunk 3]
```diff
@@ -4,11 +4,12 @@ import UpdatePassphrase from '.'
 import {compose, lifecycle, connect, type RouteProps} from '../../util/container'
 import HiddenString from '../../util/hidden-string'
 
-type OwnProps = RouteProps<{}, {}>
+type OwnProps = RouteProps<{heading: string}, {}>
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state, {routeProps}: OwnProps) => ({
```
Comment (buoyad) [inline shared/settings/passphrase/container.js]: 🔥 the `: OwnProps` here, flow should know from the type param in `connect` below

File: shared/actions/config/index.js [hunk 4]
```diff
@@ -422,7 +448,7 @@ function* configSaga(): Saga.SagaGenerator<any, any> {
   )
 
   // Like handshake but in reverse, ask sagas to do stuff before we tell the server to log us out
-  yield* Saga.chainAction<ConfigGen.LogoutPayload>(ConfigGen.logout, startLogoutHandshake)
+  yield* Saga.chainAction<ConfigGen.LogoutPayload>(ConfigGen.logout, startLogoutHandshakeIfAllowed)
```
Comment (buoyad) [inline shared/actions/config/index.js]: @chrisnojima does going through this step before starting the logout handshake look ok to you?

Comment (chrisnojima) [reply shared/actions/config/index.js]: 👍

**Pull request merged**